src/metabase/events/schema.clj

```diff
@@ -0,0 +1,133 @@
+(ns metabase.events.schema
+  (:require
+   [malli.core :as mc]
+   [toucan2.core :as t2]))
+
+;; dashboard events
+
+(def ^:private dashboard-default
+  (mc/schema
+   [:map {:closed true}
+    [:actor-id pos-int?]
+    [:object [:fn #(t2/instance-of? :model/Dashboard %)]]]))
+
+(def ^:private dashboard-update-with-dashcards
+  (mc/schema
+   [:merge
+    dashboard-default
+    [:map {:closed true}
+     [:dashcards [:sequential [:map [:id pos-int?]]]]]]))
+
+(def ^:private dashboard-update-with-tab-ids
+  (mc/schema
+   [:merge
+    dashboard-default
+    [:map {:closed true}
+     [:tab-ids [:sequential pos-int?]]]]))
+
+(def ^:private dashboard-events-schemas
+  {:event/dashboard-read             dashboard-default
+   :event/dashboard-create           dashboard-default
+   :event/dashboard-update           dashboard-default
+   :event/dashboard-reposition-cards dashboard-update-with-dashcards
+   :event/dashboard-remove-cards     dashboard-update-with-dashcards
+   :event/dashboard-add-cards        dashboard-update-with-dashcards
+   :event/dashboard-add-tabs         dashboard-update-with-tab-ids
+   :event/dashboard-update-tabs      dashboard-update-with-tab-ids
+   :event/dashboard-remove-tabs      dashboard-update-with-tab-ids})
```

Comment:
This could be much briefer
```clojure
(let [default-schema [:map {:closed true}
                      [:actor-id pos-int?]
                      [:object [:fn #(t2/instance-of? :model/Dashboard %)]]]
      with-dashcards (mut/assoc default-schema :dashcards [:sequential [:map [:id pos-int?]]])
      with-tabs      (mut/assoc default-schema :tab-ids [:sequential pos-int?])]
  (def ^:private dashboard-events-schemas
    {:event/dashboard-read             dashboard-default
     :event/dashboard-create           dashboard-default
     :event/dashboard-update           dashboard-default
     :event/dashboard-reposition-cards dashboard-update-with-dashcards
     :event/dashboard-remove-cards     dashboard-update-with-dashcards
     :event/dashboard-add-cards        dashboard-update-with-dashcards
     :event/dashboard-add-tabs         dashboard-update-with-tab-ids
     :event/dashboard-update-tabs      dashboard-update-with-tab-ids
     :event/dashboard-remove-tabs      dashboard-update-with-tab-ids}))
```

Reply:
https://github.com/metabase/metabase/pull/34947/commits/d542853d655500e2e2288514f83a1d6ec589179f